src/server/plugins/engine/pageControllers/SummaryPageController.ts

```diff
@@ -225,12 +230,54 @@ export class SummaryPageController extends PageController {
   }
 }
 
-async function sendEmail(
+async function submitForm(
   request: Request,
   summaryViewModel: SummaryViewModel,
   model: FormModel,
   state: FormSubmissionState,
   emailAddress: string
+) {
+  await extendFileRetention(model, state, emailAddress)
+  await sendEmail(request, summaryViewModel, model, emailAddress)
+}
+
+async function extendFileRetention(
+  model: FormModel,
+  state: FormSubmissionState,
+  updatedRetrievalKey: string
+) {
+  // Find out our file upload components
+  const fileUploadComponentNames = model.pages.flatMap((page) =>
+    page.components.formItems
+      .filter((item) => item.type === ComponentType.FileUploadField)
+      .map((item) => item.name)
+  )
+
+  // Pull the state for each file upload component
+  const fileUploadStates = fileUploadComponentNames.map(
+    (name) => state[name] as FileState[]
```

Comment:
`state[name]` will be null for optional file upload components - might be worth a guard

Reply:
Mind if we avoid `as` unless we're absolutely stuck and/or can't teach the compiler the types itself?

Arrays are mutable so TypeScript can intentionally not trust types when mapping

Adding a quick `Readonly<>` or `as const` can often fix this sort of thing

Reply:
Also need to consider the case when the component is nested under a section.

Something like `const pageState = this.section ? state[this.section.name] : state` or use `page.getFormDataFromState{state}`

![image](https://github.com/user-attachments/assets/d1250662-1f15-48ef-af10-e8c67b6f005e)